src/Controls/src/Xaml/XamlParser.cs

```diff
@@ -336,16 +336,25 @@ static IValueNode GetValueNode(object value, XmlReader reader)
 		static void GatherXmlnsDefinitionAttributes()
 		{
 			Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
-			s_xmlnsDefinitions = new List<XmlnsDefinitionAttribute>();
+			s_xmlnsDefinitions = [];
 
 			foreach (var assembly in assemblies)
 			{
 				try
 				{
 					foreach (XmlnsDefinitionAttribute attribute in assembly.GetCustomAttributes(typeof(XmlnsDefinitionAttribute)))
 					{
+						attribute.AssemblyName ??= assembly.FullName;
+						//maui, and x: xmlns are protected
+						if (attribute.XmlNamespace.StartsWith("http://schemas.microsoft.com/", StringComparison.Ordinal) &&
+							!attribute.AssemblyName.StartsWith("Microsoft", StringComparison.Ordinal) &&
+							!attribute.AssemblyName.StartsWith("System", StringComparison.Ordinal) &&
+							!attribute.AssemblyName.StartsWith("mscorlib", StringComparison.Ordinal))
```

Comment:
Looks lhis check is duplicated 3 times. Could you put that into a single, central place somewhere?